lib/collection/src/collection_manager/optimizers/segment_optimizer.rs

```diff
@@ -543,9 +543,7 @@ pub trait SegmentOptimizer {
 
         if !all_segments_ok {
             // Cancel the optimization
-            return Err(CollectionError::Cancelled {
-                description: "Some segments are not ready for optimization".to_string(),
-            });
+            return Ok((false, 0));
```

Comment:
This cannot be zero, because we have no segment.

```suggestion
            return Ok((false, None));
```

Reply:
But the return type is `(bool, usize)`.

Reply:
Then we need to change it :+1:

Reply:
I guess I'm a bit late to the party, by why not change it to simply `Option<SegmentId>`? As far as I can see, it's always either `(false, None)` or `(true, Some(segment_id))`. So the meaning would be: `None` - optimization cancelled/not-performed, `Some(...)` - optimization finished successfully.

Though, I don't want to prolong this anymore, so it's fine to merge as-is. We can fix this later (or never :D).